File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/BoxHelper.java
package ja.burhanrashid52.photoeditor;

import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.RelativeLayout;

/**
 * Created by Burhanuddin Rashid on 18/05/21.
 *
 * @author <https://github.com/burhanrashid52>
 */
class BoxHelper {
    private final RelativeLayout mCanvasView;
    private final PhotoEditorViewState mViewState;

    public BoxHelper(RelativeLayout canvasView, PhotoEditorViewState viewState) {
        mCanvasView = canvasView;
        mViewState = viewState;
    }

    void clearHelperBox() {
        for (int i = 0; i < mCanvasView.getChildCount(); i++) {
            View childAt = mCanvasView.getChildAt(i);
            FrameLayout frmBorder = childAt.findViewById(R.id.frmBorder);
            if (frmBorder != null) {
                frmBorder.setBackgroundResource(0);
            }
            ImageView imgClose = null;
            if (imgClose != null) {
                imgClose.setVisibility(View.GONE);
            }
        }
        mViewState.clearCurrentSelectedView();
    }

    public void clearAllViews(DrawingView drawingView) {
        for (int i = 0; i < mViewState.getAddedViewsCount(); i++) {
            mCanvasView.removeView(mViewState.getAddedView(i));
        }
        if (mViewState.containsAddedView(drawingView)) {
            mCanvasView.addView(drawingView);
        }
        mViewState.clearAddedViews();
        mViewState.clearRedoViews();

        if (drawingView != null)
            drawingView.clearAll();
    }
}

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/BoxHelper.kt
package ja.burhanrashid52.photoeditor

import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ImageView

/**
 * Created by Burhanuddin Rashid on 18/05/21.
 *
 * @author <https:></https:>//github.com/burhanrashid52>
 */
internal class BoxHelper(
    private val mViewGroup: ViewGroup,
    private val mViewState: PhotoEditorViewState
) {
    fun clearHelperBox() {
        for (i in 0 until mViewGroup.childCount) {
            val childAt = mViewGroup.getChildAt(i)
            val frmBorder = childAt.findViewById<FrameLayout>(R.id.frmBorder)
            frmBorder?.setBackgroundResource(0)
            val imgClose = childAt.findViewById<ImageView>(R.id.imgPhotoEditorClose)
            imgClose?.visibility = View.GONE
        }
        mViewState.clearCurrentSelectedView()
    }

    fun clearAllViews(drawingView: DrawingView?) {
        for (i in 0 until mViewState.addedViewsCount) {
            mViewGroup.removeView(mViewState.getAddedView(i))
        }
        drawingView?.let {
            if (mViewState.containsAddedView(it)) {
                mViewGroup.addView(it)
            }
        }

        mViewState.clearAddedViews()
        mViewState.clearRedoViews()
        drawingView?.clearAll()
    }
}